guides/documentations/essentials.md

```diff
@@ -0,0 +1,265 @@
+# How Plugin-core works
+
+Leto-modelizer is a tool that allows you to create your own infrastructure code via diagram by using a graphical interface. Behind this interface, the solution writes the code equivalent to the selected bricks (called Components) in an infrastructure language as code chosen via a plugin (Terraform, Ansible, Kubernetes resources, etc.).
+
+| Examples of diagrams |
+| :--: |
+| <img height="400" src="../../documentations/diagrams.png"/> |
+
+Plugin-core is build to contain all the base code needed for modeling tools to create those diagrams in Leto's projects. You can use this library to create your own plugin, by overriding classes and methods.
+
+Here are all classes you can use in your plugin:
+
+```js
+DefaultMetadata.js,              // Class that represents the metadata of a specific declarative code implementation.
+DefaultRender.js,                // Class that compiles a Component to data.
+DefaultParser.js,                // Class used for parsing data to Component.
+DefaultDrawer.js,                // Class used for drawing a Component in a graphical representation.
+DefaultData.js,                  // Class that represents all data of a Plugin.
+DefaultConfiguration.js,         // Class that represents default configuration of a Plugin.
+DefaultPlugin.js,                // Class that represents the default Plugin structure.
+Component.js,                    // Class that represents the base of modeling tools.
+ComponentDefinition.js,          // Class used for the definition of Component's datas and constraints.
+ComponentAttribute.js,           // Class used to add Attribute in Components.
+ComponentAttributeDefinition.js, // Class used to define Attributes for type.
+ComponentDrawOption.js,          // Class that represents option for the Component drawing.
+ComponentLink.js,                // Class that links Components together.
+ComponentLinkDefinition.js,      // Class used for the definition of the link between Components.
+ParseError.js,                   // Class that represents a parsing error. Used by the DefaultParser as default Error.
+FileInformation.js,              // Class that represents the object to store all file information.
+FileInput.js,                    // Class that represents the object to store the file content.
+Tag.js,                          // Class that represents a tag in Leto-modelizer. Used by the DefaultConfiguration class.
+Variable.js,                     // Class that represents a variable in a file of a diagram. Used by the DefaultData class.
+```
+
+| Plugin lifecycle |
+| :--: |
+| <img height="400" src="../../documentations/lifecycle.svg"/> |
+
+This is the default lifecycle of plugin usage in Leto Modelizer.
```

Comment:
After this I will add a sentence to explain plugin-core si divide in a 5 parts:
- Metadata managment
- Generate component from source 
- Render source files
- Draw diagrams
- Storage

 In each part we will explain the purpose and what is it for